recurly/resource.py

```diff
@@ -445,11 +445,23 @@ def value_for_element(cls, elem):
                 return value_class.from_element(elem)
 
         # Untyped complex elements should still be resource instances. Guess from the nodename.
-        if len(elem):  # has children
+        if len(elem) == 1:
             value_class = cls._subclass_for_nodename(elem.tag)
             log.debug("Converting %r tag into a %s", elem.tag, value_class.__name__)
             return value_class.from_element(elem)
 
+        # Untyped complex elements should still be resource instances. Guess from the nodename.
+        if len(elem) > 1:  # has childrens
+            first_tag = elem[0].tag
+            last_tag = elem[-1].tag
+
+            if(first_tag == last_tag):
+                return [cls._subclass_for_nodename(sub_elem.tag).from_element(sub_elem) for sub_elem in elem]
+            else:
+                value_class = cls._subclass_for_nodename(elem.tag)
+                log.debug("Converting %r tag into a %s", elem.tag, value_class.__name__)
+                return value_class.from_element(elem)
```

Comment:
This seems like a lot of code to add to support this new feature. There should be other resources that have similar patterns to ramps, yea? Like, what about `add_ons`/`add_on`? That's a parent/many-child relationship isn't it? What is/are the reason(s) you needed to make this change?

Reply:
looks the definition of `subscription_add_ons` in `clients/python/tests/fixtures/subscription/show.xml`, they have the array definition.

```
<subscription_add_ons type="array">
    <subscription_add_on>
        <add_on_type>usage</add_on_type>
        <measured_unit href="https://api.recurly.com/v2/measured_units/123456"/>
        <usage href="https://api.recurly.com/v2/subscriptions/123456789012345678901234567890ab/add_ons/marketing_emails/usage"/>
        <add_on_code>marketing_emails</add_on_code>
        <unit_amount_in_cents type="integer">5</unit_amount_in_cents>
        <quantity type="integer">1</quantity>
        <usage_type>price</usage_type>
        <usage_percentage nil="nil"/>
    </subscription_add_on>
  </subscription_add_ons>
```
And the ramp_intervals does not exist.
```
  <ramp_intervals>
    <ramp_interval>
      <starting_billing_cycle type="integer">1</starting_billing_cycle>
      <unit_amount_in_cents>
        <USD type="integer">2000</USD>
      </unit_amount_in_cents>
    </ramp_interval>
    <ramp_interval>
      <starting_billing_cycle type="integer">2</starting_billing_cycle>
      <unit_amount_in_cents>
        <USD type="integer">3000</USD>
      </unit_amount_in_cents>
    </ramp_interval>
  </ramp_intervals>
```

Reply:
in the file changes i left some example comments that I trying to de-serialize.